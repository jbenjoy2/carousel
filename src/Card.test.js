import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import image1 from './image1.jpg';
import image2 from './image2.jpg';
import image3 from './image3.jpg';
import Card from './Card';

it('should render', () => {
	render(
		<Card
			src={image1}
			caption="Photo by Richard Pasquarella on Unsplash"
			currNum={1}
			totalNum={1}
		/>
	);
});

it('should match snapshot', () => {
	const { asFragment } = render(
		<Card
			src={image1}
			caption="Photo by Richard Pasquarella on Unsplash"
			currNum={1}
			totalNum={1}
		/>
	);
	expect(asFragment()).toMatchSnapshot();
});

it('should match snapshot', () => {
	const { asFragment } = render(
		<Card
			src={image2}
			caption="Photo by Richard Pasquarella on Unsplash"
			currNum={2}
			totalNum={2}
		/>
	);
	expect(asFragment()).toMatchSnapshot();
});

it('should match snapshot', () => {
	const { asFragment } = render(
		<Card
			src={image3}
			caption="Photo by Richard Pasquarella on Unsplash"
			currNum={3}
			totalNum={3}
		/>
	);
	expect(asFragment()).toMatchSnapshot();
});
