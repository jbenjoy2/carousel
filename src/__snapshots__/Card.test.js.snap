// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      Photo by Richard Pasquarella on Unsplash
    </h4>
    <img
      alt="Photo by Richard Pasquarella on Unsplash"
      class="Card-image"
      src="image1.jpg"
    />
    <small
      class="Card-small"
    >
      Image 1 of 1.
    </small>
  </div>
</DocumentFragment>
`;

exports[`should match snapshot 2`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      Photo by Richard Pasquarella on Unsplash
    </h4>
    <img
      alt="Photo by Richard Pasquarella on Unsplash"
      class="Card-image"
      src="image2.jpg"
    />
    <small
      class="Card-small"
    >
      Image 2 of 2.
    </small>
  </div>
</DocumentFragment>
`;

exports[`should match snapshot 3`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      Photo by Richard Pasquarella on Unsplash
    </h4>
    <img
      alt="Photo by Richard Pasquarella on Unsplash"
      class="Card-image"
      src="image3.jpg"
    />
    <small
      class="Card-small"
    >
      Image 3 of 3.
    </small>
  </div>
</DocumentFragment>
`;
